heralding/capabilities/smtp.py

```diff
@@ -21,89 +21,251 @@
 # display, publicly perform, sublicense, relicense, and distribute [the] Contributions
 # and such derivative works.
 
-
-
+import time
+import random
+import base64
+import socket
+import asyncio
 import logging
-from base64 import b64decode
+import aiosmtpd
 
 from aiosmtpd.smtp import SMTP, MISSING, syntax
 
 from heralding.capabilities.handlerbase import HandlerBase
 
-EMPTYSTRING = ""
-logger = logging.getLogger(__name__)
+log = logging.getLogger(__name__)
+aiosmtpd.smtp.log = log
 
 
 class SMTPHandler(SMTP):
-    def __init__(self, reader, writer, options, session):
-        super().__init__(None)
-        self.transport = writer
-        self._writer = writer
+    def __init__(self, reader, writer, session, options, loop):
+        self.banner = options['protocol_specific_data']['banner']
+        super().__init__(None, hostname=self.banner, loop=loop)
+        # Reset standard banner.
+        self.__ident__ = ""
         self._reader = reader
+        self._writer = writer
+        self.transport = writer
+        self.fqdn = socket.getfqdn()
```

Comment:
This is a blocking call and can in some circumstance block the execution flow for a few seconds (happens on my setup).
Also, I am wondering if it would be more wise to either have the fqdn in settings - or set fqdn as a class variable (SMTPHandler.fqdn) at first init (maybe also update it on regular intervals using  a coroutine). Maybe even a combination of both...